build_projects/dotnet-cli-build/WriteVersionToVersionsRepo.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if !SOURCE_BUILD
+using Microsoft.Build.Framework;
+using MSBuildUtilities = Microsoft.Build.Utilities;
+using Microsoft.DotNet.VersionTools.Automation;
+using Microsoft.DotNet.VersionTools.Automation.GitHubApi;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using ThreadingTasks= System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Cli.Build
+{
+    public class WriteVersionToVersionsRepo : MSBuildUtilities.Task
+    {
+        [Required]
+        public string BranchName { get; set; }
+
+        [Required]
+        public string Name { get; set; }
+
+        [Required]
+        public string Version { get; set; }
+
+        [Required]
+        public string GitHubPassword { get; set; }
+
+        public string VersionsRepoPath { get; set; }
+        
+        public override bool Execute()
+        {
+            string versionsRepoPath = VersionsRepoPath ?? $"build-info/dotnet/core-sdk/{BranchName}";
+
+            GitHubAuth auth = new GitHubAuth(GitHubPassword);
+
+            GitHubWriteVersionUpdater repoUpdater = new GitHubWriteVersionUpdater(auth);
+            repoUpdater.UpdateBuildInfoAsync(
+                Name,
+                Version,
+                versionsRepoPath).Wait();
+
+            return true;
+        }
+    }
+
+    // This code is a slightly modified port of GitHubVersionsRepoUpdater from 
+    // https://raw.githubusercontent.com/dotnet/buildtools/master/src/Microsoft.DotNet.VersionTools/Automation/GitHubVersionsRepoUpdater.cs
+    internal class GitHubWriteVersionUpdater: VersionsRepoUpdater
```

Comment:
Could you put this in another file, or as a `private` nested class? (Don't know if's general practice, but those both seem ok to me.)